src/KubeOps/Operator/Entities/Extensions/KubernetesObjectExtensions.cs

```diff
@@ -0,0 +1,63 @@
+﻿using k8s;
+using k8s.Models;
+using Org.BouncyCastle.Asn1.Tsp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace KubeOps.Operator.Entities.Extensions
+{
+    public static class KubernetesObjectExtensions
+    {
+        public static TResource WithOwnerReference<TResource>(this TResource resource, IKubernetesObject<V1ObjectMeta> owner)
+            where TResource : IKubernetesObject<V1ObjectMeta>
+        {
+            resource.Metadata.EnsureOwnerReferences().Add(owner.MakeOwnerReference());
+            return resource;
+        }
+
+        public static IList<V1OwnerReference> EnsureOwnerReferences(this V1ObjectMeta meta)
+        {
+            if (meta.OwnerReferences == null) meta.OwnerReferences = new List<V1OwnerReference>();
+            return meta.OwnerReferences;
+        }
+
+        public static V1OwnerReference MakeOwnerReference(this IKubernetesObject<V1ObjectMeta> kubernetesObject)
+            => new V1OwnerReference(kubernetesObject.ApiVersion, kubernetesObject.Kind, kubernetesObject.Metadata.Name, kubernetesObject.Metadata.Uid);
+
+        public static V1ObjectReference MakeObjectReference(this IKubernetesObject<V1ObjectMeta> kubernetesObject)
+            => new V1ObjectReference()
+            {
+                ApiVersion = kubernetesObject.ApiVersion,
+                Kind = kubernetesObject.Kind,
+                Name = kubernetesObject.Metadata.Name,
+                NamespaceProperty = kubernetesObject.Metadata.NamespaceProperty,
+                ResourceVersion = kubernetesObject.Metadata.ResourceVersion,
+                Uid = kubernetesObject.Metadata.Uid
+            };
+
+        /// <summary>
+        /// Create an event for a kubernetesObject that can be applied to the cluster using the Create on <see cref="Client.IKubernetesClient" />
```

Comment:
Nice one, so those "events" are not directly on the kubernetes objects itself? I never had the time to inspect the mechanics around events.

So when I want to create some event about an object, I would create it with this and send it to the cluster?
The management api should then list them in `kubectl describe`?

Reply:
On the other hand, would I need to fetch an event to update the count? or does this happen magically?

Reply:
I only had a quick look into this, but they do show up in kubectl describe. I'm guessing counts I think you need to fetch and update. I cheated here by assuming one and generating a unique time based name.

Reply:
alright, then I'd suggest, we comment this out for the moment.
I'll look into events asap and see if there's a "best practice" that's given from the kubernetes team.

It would be cool to have a similar setup like "status" .

Reply:
Agree - I've commented it out for the moment